lib/ansible/modules/source_control/git.py

```diff
@@ -899,6 +917,8 @@ def main():
             recursive=dict(default='yes', type='bool'),
             track_submodules=dict(default='no', type='bool'),
             umask=dict(default=None, type='raw'),
+            archive=dict(default='no', type='bool'),
+            archive_format=dict(default="zip"),
```

Comment:
You can force choices here by adding the choices key:
`archive_format=dict(default='zip', choices=['zip', 'tar']`
By doing this, AnsibleModule will handle the argument input for you (what default is, and to fail if a bad choice is given)